projects/admin/src/app/circulation/main-request/requested-item/requested-item.component.html

```diff
@@ -0,0 +1,95 @@
+    >
+      <i class="fa fa-check pr-1"></i>
+      {{ 'Validate' | translate }}
+    </button>
+  </div>
+  <!-- FIRST ROW :: item-barcode, document-title, requester-name, request-date -->
+  <div class="col-2">
+    <button type="button" class="pl-0 pt-0 btn"
+            (click)="isCollapsed = !isCollapsed"
+            [attr.aria-expanded]="!isCollapsed"
+            aria-controls="collapse">
+      <i [ngClass]="{ 'fa-caret-down': !isCollapsed, 'fa-caret-right': isCollapsed }" class="fa" aria-hidden="true"></i>
+    </button>
+    <a [routerLink]="['/records','items','detail', item.pid]">{{ item.barcode }}</a>
+  </div>
+  <div name="title" class="col-lg-5">
+    <a [routerLink]="['/records','documents','detail', document.pid]" *ngIf="document.title | mainTitle as title ">
+      {{ title }}
+    </a><br/>
+    <ng-container *ngIf="document.contribution && document.contribution.length > 0">
+      <ul class="list-inline mb-0 small">
+        <li *ngFor="let contribution of document.contribution; let last = last; let i = index" class="list-inline-item">
+          <span *ngIf="!contribution.agent.pid" [innerHTML]="contribution | contributionFormat: true"></span>
+          <a
+            *ngIf="contribution.agent.pid"
+            [innerHTML]="contribution | contributionFormat: true"
+            [routerLink]="['/records', 'persons', 'detail', contribution.agent.pid]">
+          </a>
+          {{ last ? '' : '; ' }}
+        </li>
+      </ul>
+    </ng-container>
+  </div>
+  <div class="col-2"> {{ item.call_number }}</div>
+  <div class="col-2">{{ item.loan.transaction_date | dateTranslate :'medium' }}</div>
+  <!-- NEXT-ROWS :: request detail -->
+  <div class="col-12 mt-2" *ngIf="!isCollapsed">
+    <dl class="row">
+      <!-- item call-number -->
+      <dt class="col-sm-4 col-md-3 col-lg-2 label-title text-right" translate>Requested by</dt>
+      <dd class="col-sm-8 col-md-9 col-lg-10">
+        <ng-container *ngIf="item.loan.patron.name">
+          <a [routerLink]="['/circulation', 'patron', item.loan.patron.barcode]">{{ item.loan.patron.name }}</a>
+        </ng-container>
+      </dd>
+      <!-- item shelf location + destination -->
+      <dt class="col-sm-4 col-md-3 col-lg-2 label-title text-right" translate>Location</dt>
+      <dd class="col-sm-8 col-md-9 col-lg-10">
+        <ng-container *ngIf="item.location.pid | getRecord: 'locations' | async as location">
+          {{ location.metadata.library.pid | getRecord: 'libraries' : 'field' : 'name' | async }} &mdash;
+        </ng-container>
+        {{ item.location.name }}
+      </dd>
+      <dt class="col-sm-4 col-md-3 col-lg-2 label-title text-right" translate>Pick-up Location</dt>
+      <dd class="col-sm-8 col-md-9 col-lg-10">
+        <i class="fa fa-truck fa-flip-horizontal"></i>
+        <i class="fa fa-long-arrow-right mx-1"></i>
+        {{ item.loan.pickup_location.library_name }}
+      </dd>
+    </dl>
+  </div>
+
+</div>
+<ng-template #loading>
```

Comment:
If my memory is good we have some utilities in ng-core to manage the spinner...

Reply:
I found a `NgxSpinnerService`. This service will allow to show a spinner on the entire screen until content was loaded.
In this template, we would like to use a spinner icon (one icon for each request) until the request will be loaded.